lib/linguist/generated.rb

```diff
@@ -533,5 +534,14 @@ def generated_grpc_cpp?
       return false unless lines.count > 1
       return lines[0].start_with?("// Generated by the gRPC")
     end
+
+    # Internal: Is the file a generated Perl/Pollution/Portability header file?
+    #
+    # Returns true or false.
+    def generated_perl_ppport_header?
+        return false unless name.match(/ppport\.h$/)
+        return false unless lines.count > 10
+        return lines[8].include?("Automatically created by Devel::PPPort")
```

Comment:
Will this _always_ be on line 9?

Reply:
AFAICT, yes
https://github.com/mhx/Devel-PPPort/blame/d7849d96015b66d69606488aa65877dc2ba737a4/PPPort_pm.PL#L585

The git history goes back to 2012.

Reply:
I think in that case we can probably skip the `lines.count > 10` check.

Reply:
In Perl, if you try to perform a match against something which is undefined you will get a warning.
I don't know Ruby - does the `include` call handle array elements which don't exist?

Reply:
Okay, I just checked with a truncated file and it seemed ok.

Reply:
Ah, good point. 👍